File: frontend/types/review.ts
import { Employer, Applicant } from "./user";

export interface ReviewBase {
  id: number;
  rating: number;
  title: string;
  description: string;
}

export interface ApplicantReview extends ReviewBase { 
  employer: Employer; 
};

export interface EmployerReview extends ReviewBase { 
  reviewer: Applicant; 
};

export interface Review extends EmployerReview {
  employer: Employer; 
}

export interface ReviewFormData {
  employerId: string;
  rating: number;
  title: string;
  description: string;
}

File: backend/Backend/Models/Review.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models;

public class Review
{
	public long Id { get; set; }
	public string ReviewerId { get; set; } = string.Empty;
	public Applicant Reviewer { get; set; } = null!;
	public string EmployerId { get; set; } = string.Empty;
	public Employer Employer { get; set; } = null!;
	[Required] public double Rating { get; set; }
	[Required] [MaxLength(200)] public string Title { get; set; } = string.Empty;
	[Required] public string Description { get; set; } = string.Empty;
}
